RIOT-OS/RIOT
# sys/posix: make posix module provide only headers.

### Contribution description

The build system contains several instances of
```
 INCLUDES += -I$(RIOTBASE)/sys/posix/include
```
This is bypassing the module management system, by directly accesing headers without depending on a module. The module is the `posix` module.

That line is also added when one of the posix_* modules is requested.

[According to the docs](http://doc.riot-os.org/group__posix.html#details), the `posix` module provides headers only, but in reality there is also inet.c.

This patch:

- Moves `inet.c` into `posix_inet`, thus leaving `posix` as a headers-only module.
- Makes `posix_*` modules depend on `posix`, thus removing the explicit `INCLUDES+=...` in `sys/Makefile.include`.
- Ocurrences of `INCLUDES+=...` are replaced by an explicit dependency on `posix`.

### Testing procedure

This is a build system change, that should not change any behaviour. If I screwed up the CI should catch any modules that fail to build.
